conan/tools/system/package_manager.py

```diff
@@ -56,11 +56,12 @@ def get_default_tool(self):
                 if d in os_name:
                     return tool
 
-    def get_package_name(self, package, host_tool=True):        
+    def get_package_name(self, package, host_package=False):        
```

Comment:
I think this would be breaking wrt to existing usage:
- The current default is assuming that all packages are "host", because the architecture will be applied if cross-building, always
- Users using this might be broken by this change
- The fact that the tests have to be changed to pass, is also an indicator of this

I think we need to use the opposite default.

Reply:
But the opposite default will cause the adding of the arch atomically and it is not the native way the users use the install command. Are we fine with the fact that the default value can add the arch and do something that can feel like a magic ?

Reply:
I feel that the default behavior should be as simple as possible.
About the tests - I just think I should add tests for both cases

Reply:
> But the opposite default will cause the adding of the arch atomically and it is not the native way the users use the install command. Are we fine with the fact that the default value can add the arch and do something that can feel like a magic ?

Yes, the default behavior is intended for "host" libraries, not "build" tools. If you check all the ``system_requirements()`` in ConanCenter, you will see those are libraries intended for the "host" context. With your change, all these recipes in ConanCenter will be broken, and would require to add the ``host_package=True``. In the same way there can be tons of users with private recipes that rely on this default.

The capability that this PR is adding is the one of specifying: "I want to install a "build" tool requirement, please do not try to install it in the host architecture", which is fine to be an opt-in behavior, it doesn't seem the most popular use case so far, so the current default seems good.